src/domain/event/modal/UnenrolModal.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { FunctionComponent } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useMutation } from '@apollo/react-hooks';
+import { useHistory } from 'react-router-dom';
+
+import Modal from '../../../common/components/modal/Modal';
+import Button from '../../../common/components/button/Button';
+import unenrolOccurrenceMutation from '../mutations/unenrolOccurrenceMutation';
+
+interface UnenrolModalProps {
+  isOpen: boolean;
+  setIsOpen: (value: boolean) => void;
+  childId: string;
+  occurrenceId: string;
+}
+const UnenrolModal: FunctionComponent<UnenrolModalProps> = ({
+  isOpen,
+  setIsOpen,
+  childId,
+  occurrenceId,
+}) => {
+  const history = useHistory();
+  const { t } = useTranslation();
+
+  const [unenrolOccurrence] = useMutation(unenrolOccurrenceMutation);
+  const unenrol = async () => {
+    try {
+      await unenrolOccurrence({
+        variables: {
+          input: {
+            occurrenceId: occurrenceId,
+            childId: childId,
+          },
+        },
+      });
+      history.push('/profile');
+    } catch (error) {
+      console.error(error);
+    }
+  };
```

Comment:
See comment for https://github.com/City-of-Helsinki/kukkuu-ui/pull/198/files#diff-c33dbf914bd403da31215a71525408f8R55-R69